creme/creme_core/tests/forms/test_enumerable.py

```diff
@@ -469,26 +476,92 @@ def test_initial(self):
         self.assertListEqual(expected, [c.as_dict() for c in field.choices])
 
     def test_initial_from_model_default(self):
-        field1 = EnumerableModelChoiceField(
+        class FakeInvoiceForm(ModelForm):
+            class Meta:
+                model = FakeInvoice
+                fields = ('currency',)
+
+        form = FakeInvoiceForm()
+        field = EnumerableModelChoiceField(
             FakeInvoice, 'currency', empty_label='No value…',
         )
+        boundfield = field.get_bound_field(form, 'currency')
+
+        default_currency = Currency.objects.default()
 
-        self.assertIsNone(field1.empty_label)
-        # self.assertEqual(DEFAULT_CURRENCY_PK, field1.initial)
-        default_pk = Currency.objects.default().id
-        self.assertEqual(default_pk, field1.initial)
-        self.assertIsNone(field1.user)
+        self.assertIsNone(field.empty_label)
+        self.assertIsNone(field.user)
+        self.assertEqual(FakeInvoice._meta.get_field('currency')._get_default, field.initial)
+        self.assertEqual(default_currency.pk, boundfield.initial)
 
-        # ---
-        # other_currency = Currency.objects.exclude(id=DEFAULT_CURRENCY_PK)[0]
-        other_currency = Currency.objects.exclude(id=default_pk)[0]
-        field2 = EnumerableModelChoiceField(
+        other_currency = Currency.objects.exclude(id=default_currency.pk)[0]
+        field = EnumerableModelChoiceField(
             FakeInvoice, 'currency', empty_label='No value…', initial=other_currency.pk,
         )
+        boundfield = field.get_bound_field(form, 'currency')
+
+        self.assertIsNone(field.empty_label)
+        self.assertIsNone(field.user)
+        self.assertEqual(other_currency.pk, field.initial)
+        self.assertEqual(other_currency.pk, boundfield.initial)
+
+    @isolate_apps("creme.creme_core.tests.forms")
+    def test_initial_from_model_default__callable(self):
+        farming, industry, software = FakeSector.objects.order_by('pk')
```

Comment:
`farming, industry, software = FakeSector.objects.order_by('pk')[:3]`
(to avoid a failure if we populate more instances)